subsys/nrf_security/src/drivers/cracen/sicrypto/src/iksig.c

```diff
@@ -149,15 +148,49 @@ static int start_ecdsa_ik_sign(struct sitask *t, struct siwq *wq)
 	return SX_ERR_HW_PROCESSING;
 }
 
+static int start_ecdsa_ik_sign_wq(struct sitask *t, struct siwq *wq)
+{
+	(void)wq;
+	if (t->statuscode != SX_OK) {
+		return t->statuscode;
+	}
+	return start_ecdsa_ik_sign(t);
+}
+
```

Comment:
This is the work around to get it into the wq right? Do you then need run_ik_sign as well?

Reply:
`t->actions.run = run_ik_sign;`
the "run" action expects a function that returns "void", while start_ecdsa_ik_sign must return "int". I tried to pass start_ecdsa_ik_sign straight to actions.run, but that gave warnings.
So I guess either we use a wrapper for run, or it might be possible to cast the function return to "void" somehow. Perhaps that would look like `t->actions.run = (void)start_ecdsa_ik_sign`?

Reply:
Aaah, that makes sense. Do you need to use the `actions.run` here or could you just throw it into the wq? either of the choices lead to us discarding the return which is not optimal (Since they are all statuscodes)

Reply:
I think I will stick to actions.run to keep this similar to ecdsa sign_digest